### ISSUE TITLE
Is this useful?

### ISSUE THREAD

**Issue by hasheddan** [hasheddan/k8scr#1]
I wrote `k8scr` quickly as a way to get OCI images into registries running in Kubernetes clusters because it was relevant to work I do with [Crossplane](https://crossplane.io/). However, I imagine it could be generally useful for folks, and some have already shown interest. There are a variety of ways the scope of this project could be expanded, such as making the in-cluster registry component a little more automated (i.e. automatically deploy tiny temporary registry just to get image to whatever its final destination in the cluster is). Anyway, I am happy to expand / maintain this project if folks find it useful, so I would love to get some feedback as to what people want to see and how they may be using it already. Feel free to comment below or, if necessary, reach out privately to [@hasheddan](https://twitter.com/hasheddan).

**Comment by Vad1mo** [comment]
I am contributing to [Harbor](https://github.com/goharbor/harbor/) and there is a demand to have a distributed registry with one central unit where everyone can push to and satellites to pull images from. Something like a proxy cache and something that can distribute images. The idea is not to have a full-blown registry but something that can offload the central registry and improve the reliability of the entire system.

**Comment by hasheddan** [comment]
@Vad1mo sounds interesting! Are you imagining the "satellites" as pull-only then?

**Comment by Vad1mo** [comment]
>@Vad1mo sounds interesting! Are you imagining the "satellites" as pull-only then?

yes, but I must admit I am biased as I only focus on what customer demand, not why might be useful. There are also [P2P](https://github.com/uber/kraken) solutions out there that I didn't try yet.

IMO these are the top 3 use cases for a satellite concept:

1. Improved availability and reliability of the entire container distribution system for pulling images.
2. Less data traffic
3. Speed
